functions/_pure_prompt_nixdevshell.fish

```diff
@@ -1,13 +1,17 @@
 function _pure_prompt_nixdevshell \
-    --description "Indicate if nix develop shell is activated"
+    --description "Indicate if a nix shell is activated"
 
     if set --query pure_enable_nixdevshell;
-        and test "$pure_enable_nixdevshell" = true;
-        and test -n "$IN_NIX_SHELL"
+        and test "$pure_enable_nixdevshell" = true
 
-        set --local prefix (_pure_set_color $pure_color_nixdevshell_prefix)$pure_symbol_nixdevshell_prefix
-        set --local symbol (_pure_set_color $pure_color_nixdevshell_status)$IN_NIX_SHELL
+        set --local shell_type (_pure_prompt_nixdevshell_type)
 
-        echo "$prefix$symbol"
+        if test -n "$shell_type"
+
+            set --local prefix (_pure_set_color $pure_color_nixdevshell_prefix)$pure_symbol_nixdevshell_prefix
+            set --local symbol (_pure_set_color $pure_color_nixdevshell_status)$shell_type
+
+            echo "$prefix $symbol"
```

Comment:
We want users to control the spacing via the settings in `pure/conf.d/pure.fish `
```suggestion
            echo "$prefix$symbol"
```